statsmodels/stats/rates.py

```diff
@@ -341,19 +1108,691 @@ def tost_poisson_2indep(count1, exposure1, count2, exposure2, low, upp,
+    """power for ztest that two independent poisson rates are equal
+
+    This currently wald test type to replicate PASS chapter 436
+    analogy to proportion test.
+
+    This assumes that the variance is based on the pooled variance
+    under the null and the non-pooled variance under the alternative
+
+    Parameters
+    ----------
+    diff : float
+        Difference between rates 1 and 2 under the alternative
+    rate2 : float
+        Poisson rate for the reference sample, prop2, The rate for the
+        first sample will be computing using rate2 and diff
+        rate1 = rate2 + diff
+    nobs1 : float or int
+        number of observations in sample 1
+    nobs_ratio : float
+        sample size ratio, nobs2 = nobs_ratio * nobs1
+    alpha : float in interval (0,1)
+        Significance level, e.g. 0.05, is the probability of a type I
+        error, that is wrong rejections if the Null Hypothesis is true.
+    value : float
+        Difference between rates 1 and 2 under the null hypothesis.
+    alternative : string, 'two-sided' (default), 'larger', 'smaller'
+        Alternative hypothesis whether the power is calculated for a
+        two-sided (default) or one sided test. The one-sided test can be
+        either 'larger', 'smaller'.
+    return_results : bool
+        If true, then a results instance with extra information is returned,
+        otherwise only the computed power is returned.
+
+    Returns
+    -------
+    results : results instance or float
+        If return_results is True, then a results instance with the
+        information in attributes is returned.
+        If return_results is False, then only the power is returned.
+
+        power : float
+            Power of the test, e.g. 0.8, is one minus the probability of a
+            type II error. Power is the probability that the test correctly
+            rejects the Null Hypothesis if the Alternative Hypothesis is true.
+
+        Other attributes in results instance include :
+
+        rate_pooled
+            pooled rate, used for std_null
+        std_null
+            standard error of difference under the null hypothesis (without
+            sqrt(nobs1))
+        std_alt
+            standard error of difference under the alternative hypothesis
+            (without sqrt(nobs1))
+    """
+    # TODO: avoid possible circular import, check if needed
+    from statsmodels.stats.power import normal_power_het
+
+    diff = rate1 - rate2
+    rate_pooled, std_null, std_alt = _std_2poisson_power(
+        rate1,
+        rate2,
+        nobs_ratio=nobs_ratio,
+        alpha=alpha,
+        value=value,
+        method_var=method_var,
+        )
+
+    pow_ = normal_power_het(diff - value, nobs1, alpha, std_null=std_null,
+                            std_alternative=std_alt,
+                            alternative=alternative)
+
+    if return_results:
+        res = HolderTuple(
+            power=pow_,
+            rate_pooled=rate_pooled,
```

Comment:
rate_pooled only makes sense if value=0
now value can be also nonzero

we could use rates for "null" estimate given alternatvie, e.g. score restricted MLE.
drop for now ?